service/history/timer_queue_active_task_executor.go

```diff
@@ -678,7 +683,10 @@ func (t *timerQueueActiveTaskExecutor) executeWorkflowRunTimeoutTask(
 		return updateErr
 	}
 
-	weContext.UpdateRegistry(ctx, nil).Abort(update.AbortReasonWorkflowCompleted)
+	// A new run was created after the previous run timed out. Running Updates
+	// for this WF are aborted with a retryable error.
+	// The SDK will retry the API call, and the Update will land on the new run.
+	weContext.UpdateRegistry(ctx, nil).Abort(update.AbortReasonWorkflowContinuing)
```

Comment:
I have not exercised this code path in my testing.

Reply:
> // The SDK will retry the API call, and the Update will land on the new run.

Do we have test coverage for this claim?

Reply:
I should rephrase it because it is not only SDK but also internal server retries. They will do it first and shouldn't really reach SDK. There are tests for this retries for another CAN case, but not for timeout one. I will check if I can add one.

Reply:
I added new test.